pkg/apis/experimental/types.go

```diff
@@ -461,3 +461,75 @@ type JobCondition struct {
 	// Human readable message indicating details about last transition.
 	Message string `json:"message,omitempty"`
 }
+
+// IngressPoint encapsulates the inputs needed to connect an alias to endpoints.
+// It represents a rule collection of inbound connections from the external network
+// that would be satisfied by a load balancer.
+type IngressPoint struct {
+	api.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	api.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the IngressPoint.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Spec IngressPointSpec `json:"spec,omitempty"`
+
+	// Status is the current state of the IngressPoint.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Status IngressPointStatus `json:"status,omitempty"`
+}
+
+// IngressPointList is a collection of IngressPoints,
+type IngressPointList struct {
+	api.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	api.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of ingressPoints.
+	Items []IngressPoint `json:"items"`
+}
+
+// ServiceRef represents a reference to a single service:port.
+type ServiceRef struct {
```

Comment:
IngressBackend

On Sep 15, 2015, at 9:22 PM, Tim Hockin <notifications@github.com> wrote:

In pkg/apis/experimental/types.go
<https://github.com/kubernetes/kubernetes/pull/13947#discussion_r39585622>:

> +	Status IngressPointStatus `json:"status,omitempty"`
> +}
> +
> +// IngressPointList is a collection of IngressPoints,
> +type IngressPointList struct {
> +	api.TypeMeta `json:",inline"`
> +	// Standard object's metadata.
> +	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
> +	api.ListMeta `json:"metadata,omitempty"`
> +
> +	// Items is the list of ingressPoints.
> +	Items []IngressPoint `json:"items"`
> +}
> +
> +// ServiceRef represents a reference to a single service:port.
> +type ServiceRef struct {

or IngressServiceBackend

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/13947/files#r39585622>.